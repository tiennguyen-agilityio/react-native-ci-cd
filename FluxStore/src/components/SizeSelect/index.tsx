import {memo} from 'react';
import {StyleSheet, TouchableOpacity} from 'react-native';

// Stores
import {useThemeStore} from '@/stores';

// Components
import Flex from '../Flex';
import Text from '../Text';

const styles = StyleSheet.create({
  text: {
    textTransform: 'capitalize',
  },
});

interface SizeSelectProps {
  sizes: string[];
  defaultValue?: string;
  onValueChange: (size: string) => void;
}

const SizeSelect = ({sizes, defaultValue = '', onValueChange}: SizeSelectProps) => {
  const {
    theme: {text, background},
  } = useThemeStore();

  return (
    <Flex>
      <Text style={{color: text.tertiary}}>Size</Text>
      <Flex marginTop={10} direction="row" gap={8}>
        {sizes.map(size => {
          const isSelected = defaultValue === size;
          const handleToggleSelect = () => {
            if (size !== defaultValue) {
              onValueChange(size);
            }
          };

          return (
            <TouchableOpacity key={size} disabled={isSelected} onPress={handleToggleSelect}>
              <Flex
                justify="center"
                align="center"
                width={33}
                height={33}
                borderRadius={33}
                backgroundColor={isSelected ? background.primary : background.senary}>
                <Text style={[styles.text, {color: isSelected ? text.light : text.tertiary}]}>
                  {size}
                </Text>
              </Flex>
            </TouchableOpacity>
          );
        })}
      </Flex>
    </Flex>
  );
};

export default memo(SizeSelect);
