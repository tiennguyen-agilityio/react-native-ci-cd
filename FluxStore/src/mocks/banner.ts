export const BANNERS = [
  {
    id: '01',
    title: 'Clothing Collection 2025',
    imageUrl: 'https://i.ibb.co/wZv2s1T6/Frame-33149.png',
  },
  {
    id: '02',
    title: 'Autumn Collection 2025',
    imageUrl: 'https://i.ibb.co/hRBFHvHk/Mask-Group-1.png',
  },
  {
    id: '03',
    title: 'Accessories Collection 2025',
    imageUrl: 'https://i.ibb.co/20WCrpXW/Frame-33123.png',
  },
];
