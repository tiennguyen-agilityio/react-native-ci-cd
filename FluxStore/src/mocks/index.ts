export * from './product';
export * from './banner';
export * from './review';
export * from './cart';
