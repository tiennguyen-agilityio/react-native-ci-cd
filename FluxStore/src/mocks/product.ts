import {Product} from '@/interfaces';

export const COLORS = ['black', 'red', 'blue'];
export const SIZES = ['S', 'M', 'L'];

export const PRODUCTS: Product[] = [
  {
    id: '01',
    name: 'Linen Dress',
    price: 90.02,
    discount: 10,
    rating: 5,
    reviewCount: 20,
    image: 'https://i.ibb.co/SD47BZNp/Mask-Group-1.png',
    description:
      'Minimal Stand is made of by natural wood. The design that is very simple and minimal. This is truly one of the best furnitures in any family for now. With 3 different colors, you can easily select the best match for your home.',
    sizes: ['s', 'M'],
    colors: ['black', 'red', 'blue'],
    carouselImages: [
      'https://i.ibb.co/KQj6DyL/image-87.png',
      'https://i.ibb.co/SD47BZNp/Mask-Group-1.png',
      'https://i.ibb.co/Jw9tjsj3/Mask-Group-2.png',
    ],
  },
  {
    id: '02',
    name: 'Filted Waist Dress',
    price: 82.0,
    discount: 0,
    rating: 3,
    reviewCount: 110,
    image: 'https://i.ibb.co/Jw9tjsj3/Mask-Group-2.png',
    description:
      'Minimal Stand is made of by natural wood. The design that is very simple and minimal. This is truly one of the best furnitures in any family for now. With 3 different colors, you can easily select the best match for your home.',
    sizes: ['XL'],
    colors: ['pink', 'black'],
    carouselImages: [
      'https://i.ibb.co/KQj6DyL/image-87.png',
      'https://i.ibb.co/SD47BZNp/Mask-Group-1.png',
      'https://i.ibb.co/Jw9tjsj3/Mask-Group-2.png',
    ],
  },
];
