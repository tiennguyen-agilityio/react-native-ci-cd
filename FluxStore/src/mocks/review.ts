export const REVIEWS = [
  {
    id: '01',
    avatarUrl: 'https://i.ibb.co/h1n0FRHn/Ellipse-76.png',
    name: 'Jennifer Rose',
    rating: 5,
    comment:
      'I love it.  Awesome customer service!! Helped me out with adding an additional item to my order. Thanks again!',
    createdAt: 1747710042553,
  },
  {
    id: '02',
    avatarUrl: 'https://i.ibb.co/tT8QbLVp/Ellipse-77.png',
    name: 'Kelly Rihana',
    rating: 5,
    comment: "I'm very happy with order, It was delivered on and good quality. Recommended!",
    createdAt: 1747710042570,
  },
];
