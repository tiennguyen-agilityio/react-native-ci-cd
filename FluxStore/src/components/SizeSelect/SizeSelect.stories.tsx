import {Meta, StoryObj} from '@storybook/react';
import {Alert} from 'react-native';

import {SIZES} from '@/mocks';
import SizeSelect from '.';

const meta: Meta<typeof SizeSelect> = {
  title: 'Components/SizeSelect',
  component: SizeSelect,
};

export default meta;

type Story = StoryObj<typeof SizeSelect>;

const defaultProps = {
  sizes: SIZES,
  defaultValue: SIZES[0],
  onValueChange: (color: string) => Alert.alert('Current sizes', color),
};

export const Default: Story = {
  args: defaultProps,
};
