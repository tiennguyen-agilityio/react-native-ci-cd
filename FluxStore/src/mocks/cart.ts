import {PRODUCTS} from './product';

export const CARTS = [
  {
    id: '01',
    product: {
      id: PRODUCTS[0].id,
      image: PRODUCTS[0].image,
      name: PRODUCTS[0].name,
      price: PRODUCTS[0].price,
    },
    sizes: [PRODUCTS[0]?.sizes?.[0], PRODUCTS[0]?.sizes?.[1]].join(', '),
    colors: [PRODUCTS[0]?.colors?.[0]].join(', '),
    quantity: 2,
    isChecked: true,
  },
  {
    id: '02',
    product: {
      id: PRODUCTS[1].id,
      image: PRODUCTS[1].image,
      name: PRODUCTS[1].name,
      price: PRODUCTS[1].price,
    },
    sizes: [PRODUCTS[1]?.sizes?.[0]].join(', '),
    colors: [PRODUCTS[1]?.colors?.[0], PRODUCTS[1]?.colors?.[1]].join(', '),
    quantity: 1,
    isChecked: false,
  },
];
