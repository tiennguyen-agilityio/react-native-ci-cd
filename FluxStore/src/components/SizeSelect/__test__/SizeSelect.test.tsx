import {fireEvent, render, screen} from '@testing-library/react-native';

import {SIZES} from '@/mocks';
import SizeSelect from '..';

const onValueChange = jest.fn();
const props = {
  sizes: SIZES,
  onValueChange,
};

describe('SizeSelect Component', () => {
  it('renders correctly', () => {
    const {toJSON} = render(<SizeSelect {...props} />);

    expect(toJSON()).toMatchSnapshot();
  });

  it('renders correctly with defaultIndex', () => {
    const {toJSON} = render(<SizeSelect {...props} defaultValue={SIZES[0]} />);

    expect(toJSON()).toMatchSnapshot();
  });

  it('renders all sizes and handles selection correctly', () => {
    render(<SizeSelect {...props} />);

    // Check all sizes render
    SIZES.forEach(size => {
      expect(screen.getByText(size)).toBeTruthy();
    });

    // Simulate pressing 'M'
    fireEvent.press(screen.getByText('M'));

    // Should call onValueChange with ['M']
    expect(onValueChange).toHaveBeenCalledWith('M');

    // Press again to deselect
    fireEvent.press(screen.getByText('M'));

    // Should call onValueChange with []
    expect(onValueChange).toHaveBeenCalledWith('M');
  });
});
