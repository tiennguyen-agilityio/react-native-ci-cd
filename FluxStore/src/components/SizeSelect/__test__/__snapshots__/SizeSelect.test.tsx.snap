// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SizeSelect Component renders correctly 1`] = `
<View
  style={
    [
      {
        "flexDirection": "column",
      },
      {
        "justifyContent": "flex-start",
      },
      {
        "alignItems": "stretch",
      },
      {
        "flexWrap": "nowrap",
      },
      undefined,
      {},
    ]
  }
  testID="flex"
>
  <Text
    style={
      [
        {
          "letterSpacing": 0,
          "textAlign": "left",
          "textAlignVertical": "center",
        },
        {
          "color": "#737680",
          "fontFamily": "ProductSansRegular",
          "fontSize": 14,
        },
        {},
        {
          "color": "#777e90",
        },
      ]
    }
  >
    Size
  </Text>
  <View
    style={
      [
        {
          "flexDirection": "row",
        },
        {
          "justifyContent": "flex-start",
        },
        {
          "alignItems": "stretch",
        },
        {
          "flexWrap": "nowrap",
        },
        undefined,
        {
          "gap": 8,
          "marginTop": 10,
        },
      ]
    }
    testID="flex"
  >
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": false,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "opacity": 1,
        }
      }
    >
      <View
        style={
          [
            {
              "flexDirection": "column",
            },
            {
              "justifyContent": "center",
            },
            {
              "alignItems": "center",
            },
            {
              "flexWrap": "nowrap",
            },
            undefined,
            {
              "backgroundColor": "#f3f3f3",
              "borderRadius": 33,
              "height": 33,
              "width": 33,
            },
          ]
        }
        testID="flex"
      >
        <Text
          style={
            [
              {
                "letterSpacing": 0,
                "textAlign": "left",
                "textAlignVertical": "center",
              },
              {
                "color": "#737680",
                "fontFamily": "ProductSansRegular",
                "fontSize": 14,
              },
              {},
              [
                {
                  "textTransform": "capitalize",
                },
                {
                  "color": "#777e90",
                },
              ],
            ]
          }
        >
          S
        </Text>
      </View>
    </View>
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": false,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "opacity": 1,
        }
      }
    >
      <View
        style={
          [
            {
              "flexDirection": "column",
            },
            {
              "justifyContent": "center",
            },
            {
              "alignItems": "center",
            },
            {
              "flexWrap": "nowrap",
            },
            undefined,
            {
              "backgroundColor": "#f3f3f3",
              "borderRadius": 33,
              "height": 33,
              "width": 33,
            },
          ]
        }
        testID="flex"
      >
        <Text
          style={
            [
              {
                "letterSpacing": 0,
                "textAlign": "left",
                "textAlignVertical": "center",
              },
              {
                "color": "#737680",
                "fontFamily": "ProductSansRegular",
                "fontSize": 14,
              },
              {},
              [
                {
                  "textTransform": "capitalize",
                },
                {
                  "color": "#777e90",
                },
              ],
            ]
          }
        >
          M
        </Text>
      </View>
    </View>
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": false,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "opacity": 1,
        }
      }
    >
      <View
        style={
          [
            {
              "flexDirection": "column",
            },
            {
              "justifyContent": "center",
            },
            {
              "alignItems": "center",
            },
            {
              "flexWrap": "nowrap",
            },
            undefined,
            {
              "backgroundColor": "#f3f3f3",
              "borderRadius": 33,
              "height": 33,
              "width": 33,
            },
          ]
        }
        testID="flex"
      >
        <Text
          style={
            [
              {
                "letterSpacing": 0,
                "textAlign": "left",
                "textAlignVertical": "center",
              },
              {
                "color": "#737680",
                "fontFamily": "ProductSansRegular",
                "fontSize": 14,
              },
              {},
              [
                {
                  "textTransform": "capitalize",
                },
                {
                  "color": "#777e90",
                },
              ],
            ]
          }
        >
          L
        </Text>
      </View>
    </View>
  </View>
</View>
`;

exports[`SizeSelect Component renders correctly with defaultIndex 1`] = `
<View
  style={
    [
      {
        "flexDirection": "column",
      },
      {
        "justifyContent": "flex-start",
      },
      {
        "alignItems": "stretch",
      },
      {
        "flexWrap": "nowrap",
      },
      undefined,
      {},
    ]
  }
  testID="flex"
>
  <Text
    style={
      [
        {
          "letterSpacing": 0,
          "textAlign": "left",
          "textAlignVertical": "center",
        },
        {
          "color": "#737680",
          "fontFamily": "ProductSansRegular",
          "fontSize": 14,
        },
        {},
        {
          "color": "#777e90",
        },
      ]
    }
  >
    Size
  </Text>
  <View
    style={
      [
        {
          "flexDirection": "row",
        },
        {
          "justifyContent": "flex-start",
        },
        {
          "alignItems": "stretch",
        },
        {
          "flexWrap": "nowrap",
        },
        undefined,
        {
          "gap": 8,
          "marginTop": 10,
        },
      ]
    }
    testID="flex"
  >
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": true,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={false}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "opacity": 1,
        }
      }
    >
      <View
        style={
          [
            {
              "flexDirection": "column",
            },
            {
              "justifyContent": "center",
            },
            {
              "alignItems": "center",
            },
            {
              "flexWrap": "nowrap",
            },
            undefined,
            {
              "backgroundColor": "#343434",
              "borderRadius": 33,
              "height": 33,
              "width": 33,
            },
          ]
        }
        testID="flex"
      >
        <Text
          style={
            [
              {
                "letterSpacing": 0,
                "textAlign": "left",
                "textAlignVertical": "center",
              },
              {
                "color": "#737680",
                "fontFamily": "ProductSansRegular",
                "fontSize": 14,
              },
              {},
              [
                {
                  "textTransform": "capitalize",
                },
                {
                  "color": "#fff",
                },
              ],
            ]
          }
        >
          S
        </Text>
      </View>
    </View>
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": false,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "opacity": 1,
        }
      }
    >
      <View
        style={
          [
            {
              "flexDirection": "column",
            },
            {
              "justifyContent": "center",
            },
            {
              "alignItems": "center",
            },
            {
              "flexWrap": "nowrap",
            },
            undefined,
            {
              "backgroundColor": "#f3f3f3",
              "borderRadius": 33,
              "height": 33,
              "width": 33,
            },
          ]
        }
        testID="flex"
      >
        <Text
          style={
            [
              {
                "letterSpacing": 0,
                "textAlign": "left",
                "textAlignVertical": "center",
              },
              {
                "color": "#737680",
                "fontFamily": "ProductSansRegular",
                "fontSize": 14,
              },
              {},
              [
                {
                  "textTransform": "capitalize",
                },
                {
                  "color": "#777e90",
                },
              ],
            ]
          }
        >
          M
        </Text>
      </View>
    </View>
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": false,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "opacity": 1,
        }
      }
    >
      <View
        style={
          [
            {
              "flexDirection": "column",
            },
            {
              "justifyContent": "center",
            },
            {
              "alignItems": "center",
            },
            {
              "flexWrap": "nowrap",
            },
            undefined,
            {
              "backgroundColor": "#f3f3f3",
              "borderRadius": 33,
              "height": 33,
              "width": 33,
            },
          ]
        }
        testID="flex"
      >
        <Text
          style={
            [
              {
                "letterSpacing": 0,
                "textAlign": "left",
                "textAlignVertical": "center",
              },
              {
                "color": "#737680",
                "fontFamily": "ProductSansRegular",
                "fontSize": 14,
              },
              {},
              [
                {
                  "textTransform": "capitalize",
                },
                {
                  "color": "#777e90",
                },
              ],
            ]
          }
        >
          L
        </Text>
      </View>
    </View>
  </View>
</View>
`;
